### PR TITLE
Logging of errors thrown by evaluated JS and improments

### PR DESCRIPTION
**Pull request by N0taN3rd**
added additional documentation that was lacking
properly sorted the python imports
added logging of errors thrown by JS that a tab evaluates in the page
reverted back to using dict literal rather than constructor
tweaked detection of navigation to actual pages in crawler tab by only looking for "html" in the mime type of the navigation response
fixed issue where the currently crawled URL is not updated in redis by pulling out the logic for determining the next crawler action based on the navigation results into its own method
imported handling of adding outlinks to the frontier in order to improve the chances that duplicates URLs are not added by a crawler in the same automation

### CODE REVIEW

File: autobrowser/util/helper.py [hunk 1]
```diff
@@ -7,23 +7,22 @@
...
 
 ListenerDict = Dict[str, Union[str, EventEmitter, Callable]]
 
 
-class Helper:
-    """Utility class providing helpful utility functions"""
-
-    __slots__ = ()
+class Helper(SC_Helper):
+    """Utility class providing helpful utility functions that extends the helper classed by simplechrome"""
```
Comment (ikreymer) [inline autobrowser/util/helper.py]: 'classed' typo

File: autobrowser/tabs/crawlerTab.py [hunk 2]
```diff
@@ -110,56 +123,61 @@ def wait_for_net_idle(
...
 
     async def crawl(self) -> None:
+        """Starts the crawl loop.
+
+        For each URL in the frontier:
+          - navigate to the page
+          - perform the next crawler action based on the navigation results
+
+        The crawl loop is exited once the frontier becomes exhausted or one of the following conditions are met:
+           - shutdown, gracefull or otherwise, is initiated
```
Comment (ikreymer) [inline autobrowser/tabs/crawlerTab.py]: 'gracefull' typo

File: autobrowser/frontier/redis.py [hunk 3]
```diff
@@ -116,72 +135,78 @@ def remove_from_pending(self, url: str) -> Awaitable[Any]:
...
+        if is_seen:
+            self.logger.info(
+                logged_method, f"Not adding URL to the frontier, seen - {url_info}"
             )
+            return False
+
+        self.logger.info(logged_method, f"Adding URL to the frontier - {url_info}")
+        await aio_gather(
+            self.redis.rpush(self.keys.queue, ujson_dumps(url_info)),
+            self.redis.sadd(self.keys.seen, url),
```
Comment (ikreymer) [inline autobrowser/frontier/redis.py]: `sadd` returns 1 if added or 0 if already present.. perhaps should have this call rpush after sadd and only if it returns 1?

Comment (N0taN3rd) [reply autobrowser/frontier/redis.py]: we take care of the seen check on a few lines before
`is_seen = await self.is_seen(url)`

**Pull request merged**